README.md

```diff
@@ -2,29 +2,26 @@
 [![codecov](https://codecov.io/gh/opensearch-project/search-processor/branch/main/graph/badge.svg?token=PYQO2GW39S)](https://codecov.io/gh/opensearch-project/search-processor)
 ![PRs welcome!](https://img.shields.io/badge/PRs-welcome!-success)
 
-# Search Query & Request Transformers
+# Search Rerankers: AWS Kendra & AWS Personalize
 - [Welcome!](#welcome)
 - [Project Resources](#project-resources)
 - [Code of Conduct](#code-of-conduct)
 - [License](#license)
 - [Copyright](#copyright)
 
 ## Welcome!
-This repository is the home of an evolving project that aims to create a pipeline of transformers to preprocess queries before search and post-process results after search. The first component here is a plugin to re-rank search results before returning them to the client inline. In the coming year, we will add hooks to configure other re-rankers and allow users to add their own components to the pipeline. Logging will also be a critical part of the pipeline in two ways: 
-1. Logging information about the search experience (e.g. query, search results returned from the index, search results returned to the OpenSearch client) 
-1. Logging debug information about the transformers
+This repository hosts the code for two self-install re-rankers that integrate into [Search Pipelines](https://opensearch.org/docs/latest/search-plugins/search-pipelines/index/). User documentation for the Personalize Reranker is [here](https://opensearch.org/docs/latest/search-plugins/search-pipelines/personalize-search-ranking/). For Kendra, it is [here](https://opensearch.org/docs/latest/search-plugins/search-relevance/index/#reranking-results-with-kendra-intelligent-ranking-for-opensearch). The current guideline for developing processors is that if you are developing a processor that makes a network connection outside of OpenSearch, it should be in a separate repository. If it does not make a separate connection, it could go into the OpenSearch repository under [org.opensearch.search.pipeline.common](https://github.com/opensearch-project/OpenSearch/tree/a08d588691c3b232e65d73b0a0c2fc5c72c870cf/modules/search-pipeline-common). Please create a PR there if you have an idea for a new processor.
```

Comment:
>The current guideline for developing processors is that if you are developing a processor that makes a network connection outside of OpenSearch, it should be in a separate repository

If someone wishes to add a new processor argument search results by obtaining data from a Spark cluster, it may be a broad data source processor or a very customized Spark processor implementation. Do they both fall within this category? Should we advise that the new processor reside in this repo or in a separate repo in the opensearch-project?

Reply:
That's a good point. Can you expand on the difference between a "broad data source processor vs customized Spark processor?" I would not want to put any specific business logic in core. I wouldn't even want to put that in a separate repo. A Spark job definition (via SQL or a specific implementation in a JVM language (or I suppose Python)) should be passed to the processor as a parameter. 

We may look to the connectors framework under development to define this better. There is one implementation (Apache Spark) and at least a few deployments of Spark (e.g. Databricks, EMR). I'm not sure if the way you connect to each is different, but I think I would want to leave that detail out of a generic Spark processor. 

More concretely, the DynamoDB processor could be a good example. Because it's a specific database implementation I would want to see much of that outside of core, but there may be helpers or other ways to make it easier for developers to create any number of processors that pull data from a database. Are there similarities between Cassandra, DynamoDB, Oracle, MongoDB, MySQL, and even Spark when it comes to processors that we might factor out and put in core?

What I do believe is that no more processors belong here. Do we agree on that? We can always ask for PRs against core if people want to create a processor. Maintainers can then judge whether the processor belongs there or somewhere else.

cc: @msfroh

Reply:
Rethinking this a bit: we should consider guidelines in terms of dependencies. Introducing new dependencies into @opensearch-project/OpenSearch should be discouraged. In the case of Spark, I would say that if there are abstractions that could make it easier to bring a processor in that access external data sources, we should consider those in core, but I would have a hard time adding a dependency on Spark in core for almost any reason. Will reflect this in the next revision.